Rework GT++ Tab to "Mass Processing"

FourIsTheNumber opened the pull request:
This is a minor-ish rework of the GT++ tab. Instead of being based around GT++ in particular, it is now a tab dedicated to multiblocks that scale your production (multiblocks which are stronger versions of singleblock or even multiblock machines). The vast majority of the quests in the GT++ tab fell under this category already - those which did not were simply moved to other tabs.

I have also removed some explicit references to GT++ in quest names and descriptions in order to fit the new tab.

The Mass Processing tab has several new quests:
-An introduction to multiblock processing that explains terms like "Parallel", "Overclock", "Perfect Overclock". This is right before the steam multi quests.
-An introduction to megas. I have also changed the mega quests to no longer require IV or give IV lootbags - these multis are available in HV and locking the quests this way is strange.
-Quest for the mega oil cracker, because there wasn't one.
-An introduction to HV Mass Processing multis.
-Three new quests for multis I have introduced recently.

### Affected tabs:
This is the new state of the "Mass Processing" tab.
![2024-08-20_15 09 32](https://github.com/user-attachments/assets/503f9352-1cee-4e94-81df-68a85053f7a1)

Multiblock Goals has had a few multiblocks moved in and out of it. Crucial progression multis like the QFT, Cyclotron, and Chemical Plant have been moved here. LCR, Distillation Tower, AAL, and all Megas were moved to Mass Processing.
![2024-08-20_15 10 29](https://github.com/user-attachments/assets/2adc0d30-f45e-4225-9837-eb549b1953fe)

How to Generate Power now has the Solar Tower. I have also moved the plasma chain here - it did not make much sense in GT++ originally and it makes less sense now. In the future I would like to move this into a dedicated fusion tab along with some other chains.
![2024-08-19_19 42 52](https://github.com/user-attachments/assets/4f5f78f6-e796-453c-bd7a-d6544d4c90ea)

Storing and Transforming Power has the Energy Buffer and Wireless Charger quests.
![2024-08-19_19 43 08](https://github.com/user-attachments/assets/2d90d974-da34-49a9-ba78-8817c3b50e6f)

### New Quests:

![2024-08-20_13 13 31](https://github.com/user-attachments/assets/bdecba8f-655e-48cd-b047-e2e242044fc3)
![2024-08-19_19 50 46](https://github.com/user-attachments/assets/f41eba60-9c0d-438f-928e-d15972fd24ab)
![2024-08-19_19 50 58](https://github.com/user-attachments/assets/9e8829da-c382-41ad-b93e-b8d00ef4a07d)
![2024-08-19_19 51 12](https://github.com/user-attachments/assets/2dac67d9-3ab7-473e-b580-438008801114)
![2024-08-19_19 51 33](https://github.com/user-attachments/assets/dcdf28d2-3ecd-431a-955f-2be9621ee91e)
![2024-08-19_19 51 49](https://github.com/user-attachments/assets/67219784-b324-4c4c-86c6-6e26699da332)
![2024-08-19_19 52 01](https://github.com/user-attachments/assets/6fe1b85b-aac8-452f-88c0-3890adf281fc)

One quest was removed: The quest for GT++ books. I have moved these to be optional tasks within their respective multis instead.

For more info on my thoughts behind doing this: refer to this proposal:
https://excalidraw.com/#json=7c0zACjrOjFdIuKSOdFW4,CTpziN8VCXl6mbsuy6rbag

as well as some discussions in #quest-dev on the discord.

TLDR: This will help to make the pack a more cohesive experience - instead of randomly singling out one GT addon for its own tab, keep multis sorted by their impact on players.

Note that even more multis are soon to be added that would fit right in here :)

Baerschen commented:
Just downloaded your branch and looked into it ingame and i must say this looks pretty good. At first I thought it would look cluttered as i mentioned on discord but thats actually not the case. This makes it way better for adding new multis aswell down the line. Very well done! On top of that explaining mechanics like parallel and perfect OC is very well done and much needed information for new players.

chochem reviewed (changes requested):
heavily dependent on https://github.com/GTNewHorizons/GT5-Unofficial/pull/2799
as long as that is the case you will need to wait until that is fully resolved

chochem commented:
also something else: 
the start of the tab is very broken right now. "multiblocks in steam age? impossible" was forgotten. while multiblock revolution doesnt have a prereq. those should probably be one and the same quest on the old id and in both tabs.

Dream-Master commented:
> heavily dependent on [GTNewHorizons/GT5-Unofficial#2799](https://github.com/GTNewHorizons/GT5-Unofficial/pull/2799) as long as that is the case you will need to wait until that is fully resolved

this pr is done and can be reviewed

Baerschen commented:
> also something else: the start of the tab is very broken right now. "multiblocks in steam age? impossible" was forgotten. while multiblock revolution doesnt have a prereq. those should probably be one and the same quest on the old id and in both tabs.

Yeah the quest in LV tab tells you to look at the GT++ Tab so that needs to be adjusted. Would also add the information that multis void on power loss to the revolution quest and remove it from the central steam quest and potentially other central quests.

chochem commented on config/betterquesting/DefaultQuests/Quests/HowtoGeneratePow-AAAAAAAAAAAAAAAAAAAACQ==/DeepEarthHeating-AAAAAAAAAAAAAAAAAAAKzg==.json:
```diff
@@ -50,9 +50,9 @@
       "requiredItems:9": {
         "0:10": {
-          "id:8": "gregtech:gt.blockmachines",
+          "id:8": "gregtech:gt.blockframes",
```
here is another

chochem commented on config/betterquesting/DefaultQuests/Quests/MultiblockGoals-AAAAAAAAAAAAAAAAAAAADA==/DiggingDeepforWa-jreOZVXRQa2sGXViV7-3Gg==.json:
```diff
@@ -52,9 +52,9 @@
         },
         "1:10": {
-          "id:8": "gregtech:gt.blockmachines",
+          "id:8": "gregtech:gt.blockframes",
```
and here

FourIsTheNumber commented:
> > also something else: the start of the tab is very broken right now. "multiblocks in steam age? impossible" was forgotten. while multiblock revolution doesnt have a prereq. those should probably be one and the same quest on the old id and in both tabs.
> 
> Yeah the quest in LV tab tells you to look at the GT++ Tab so that needs to be adjusted. Would also add the information that multis void on power loss to the revolution quest and remove it from the central steam quest and potentially other central quests.

Merged these two quests, updated the text, and shared them between the tabs. Moved power loss info as well
